src/metrics/metrics.md

```diff
@@ -1,11 +1,62 @@
 # Metrics
 
-Here we provide definitions and information about our standard metrics.
+Here we provide definitions and information about our standard metrics.  We use some terminology that is defined [here](https://docs.telemetry.mozilla.org/concepts/terminology.html).
 
-Table of contents with links to all metrics.
+## Table of Contents
+
+<!-- toc -->
+
+-----
+
+## Daily Active Users (DAU)
+
+### Overview
+
+Daily Active Users or DAU counts the number of unique profiles active in the product on each day.  This is intended to approximate the number of people using the product each day.
+
+### Details
+
+The concept of a "unique profile" may vary from product to product and the relationship between that concept and actual users may also vary.  As such care should be taken interpreting this metric as an actual count of users.
+
+The concept of being "active" is defined as in the [`usage criterion`](./usage.md) section and will vary by product.  A product may even have multiple possible `usage criteria`, but there will be one default choice that is our best approximation of general product usage.
```

Comment:
I think it's a bad idea to include usage criteria here. I think that the industry standard is that "DAU" is a concrete metric, with "Active" having some fixed definition that's relevant to the product.

I think we're asking for trouble if we define "DAU" as a parametrised metric. Most Mozilla staff are going to talk about "Desktop DAU" rather than "Desktop DAU with the usage criterion of sending a ping". This would mean that their usage of the terminology would be technically incorrect, and underspecified.

For what benefit? So that we could say "Desktop DAU with the usage criterion of having nonzero active hours", instead of "The number of desktop clients per day with nonzero active hours"? Is there any advantage to bringing the "DAU" acronym in here and diluting it?


I understand that in the context of building GUD, it made sense to write one query/function that solves a class of problems, rather than many queries/functions that each solve a narrow problem. In the context of writing a tool, or designing a UI, you want a small set of things that can combine to solve a large class of problems.

But here in the context of defining key metrics, I claim that we don't want to define broad classes of metrics: we want to define the key metrics that people are watching and making decisions from. And we don't want to leave room for ambiguity

Reply:
GUD has become the standard tool for measuring DAU and explicitly uses usage criteria.  I agree that we want a "standard DAU" per product, but allowing DAU to be in terms of alternative usage criteria doesn't seem to be asking too much of our users.

I tried to make this clear in the text "there will be one default choice that is our best approximation of general product usage" and that if we just say Desktop DAU, that is well-defined.  Do you think the text isn't clear enough on this point?

Reply:
> allowing DAU to be in terms of alternative usage criteria doesn't seem to be asking too much of our users.

It's asking too much of me! It means that I need to say "DAU for sends a ping" instead of "DAU", because the latter would be undefined without a usage criterion.

> I tried to make this clear in the text "there will be one default choice that is our best approximation of general product usage" and that if we just say Desktop DAU, that is well-defined. Do you think the text isn't clear enough on this point?

Yes, I think it's unclear: nowhere in this document does it specify what that default choice is! Someone could carefully comb this document, which seeks to define DAU, and they would be unable to learn that Firefox Desktop DAU is the number of profiles for which a ping is received on a given day!

And to be clear, I think the best solution for this document is to define "DAU" with a single usage criterion per platform, rather than to parametrise it and specify that default parameter somewhere in the document.

Reply:
I think we're thinking along the same lines and maybe just need to make the text better reflect what we're thinking.

"DAU" should mean a particular thing per product.  But that thing may not always be "sent a ping".  We already have some confusion around "which ping" for Fenix and, although this is not the place to discuss it, I hope we move to requiring non-zero active hours or some similar restriction for at least desktop in the future.

We don't want to have to have a separate DAU metric defined for each product.  Which is why we can have a single parameterized definition, while making clear that there is a standard parameter for each platform.

So maybe the text should make clear that DAU counts active users of a product and how we define active use is in the usage criteria section.

Sound fair?

Reply:
I think this definition is not complete unless it includes the usage criteria for each platform, on this page. Otherwise this page does not give a complete answer to the basic question "what is Firefox Desktop DAU?"! Shouldn't this page answer that question without sending people on an expedition?

Reply:
I guess the only point of disagreement now is whether to list all the standard usage criteria with the metric description.  If we do so, we're repeating the list of criteria on many metric descriptions, as they're needed to DAU. MAU, WAU, Intensity, probably Retention, and many others.  Doesn't it seem cleaner to have a separate list of usage criteria?  Asking a user to look at two pages to thoroughly understand a metric doesn't seem terrible.